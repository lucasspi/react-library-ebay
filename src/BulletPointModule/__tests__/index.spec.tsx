import React from "react";
import { cleanup, render } from "@testing-library/react";
import Module from "../index";
import { defaultModule } from "./fixtures";

const component = (data = defaultModule) => render(<Module data={data} />);
let data = defaultModule;

describe(`<Module {...defaultProps} />`, () => {
  afterEach(cleanup);

  it("should render steps with default icons given a model without Experience Icon", () => {
    const { asFragment } = render(<Module data={defaultModule} />);
    expect(asFragment()).toMatchSnapshot();
  });

  describe(`Should not be render`, () => {
    it("should return empty HTML Element if step is undefined", () => {
      data.steps = undefined;
      expect(component(data).container).toBeEmptyDOMElement();
    });

    it("should not render steps if steps is empty", () => {
      data.steps = [];
      expect(component(data).container).not.toHaveClass(".step");
    });
  });
});
