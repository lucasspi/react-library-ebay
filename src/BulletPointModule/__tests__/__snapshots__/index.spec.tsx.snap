// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Module {...defaultProps} /> should render steps with default icons given a model without Experience Icon 1`] = `
<DocumentFragment>
  <div
    class="bullet-steps"
  >
    <h2>
      <span
        class="eui-textual-display"
      >
        <span
          class="eui-text-span "
        >
          <span
            class="BOLD"
          >
            Hereâ€™s what you can get
          </span>
        </span>
      </span>
    </h2>
    <div
      class="step-list"
    >
      <div
        class="step"
      >
        <div>
          <div
            class="bullet-point"
          >
            1
          </div>
        </div>
        <div
          class="text-step"
        >
          <h3>
            <span
              class="eui-textual-display"
            >
              <span
                class="eui-text-span "
              >
                <span
                  class="BOLD"
                >
                  All-in-one management
                </span>
              </span>
            </span>
          </h3>
          <span
            class="eui-textual-display"
          >
            <span
              class="eui-text-span "
            >
              <span
                class="SECONDARY"
              >
                Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc eleifend arcu tincidunt gravida auctor. Vivamus nec velit sed mi ultrices bibendum.
              </span>
            </span>
          </span>
        </div>
      </div>
      <div
        class="step"
      >
        <div>
          <div
            class="bullet-point"
          >
            2
          </div>
        </div>
        <div
          class="text-step"
        >
          <h3>
            <span
              class="eui-textual-display"
            >
              <span
                class="eui-text-span "
              >
                <span
                  class="BOLD"
                >
                  Price guidance and sales history
                </span>
              </span>
            </span>
          </h3>
          <span
            class="eui-textual-display"
          >
            <span
              class="eui-text-span "
            >
              <span
                class="SECONDARY"
              >
                Sed vel lacus ut ante lacinia vehicula. Cras hendrerit posuere sapien eget pulvinar. Nam dictum velit sed augue elementum venenatis.
              </span>
            </span>
          </span>
        </div>
      </div>
      <div
        class="step"
      >
        <div>
          <div
            class="bullet-point"
          >
            3
          </div>
        </div>
        <div
          class="text-step"
        >
          <h3>
            <span
              class="eui-textual-display"
            >
              <span
                class="eui-text-span "
              >
                <span
                  class="BOLD"
                >
                  Instant listing
                </span>
              </span>
            </span>
          </h3>
          <span
            class="eui-textual-display"
          >
            <span
              class="eui-text-span "
            >
              <span
                class="SECONDARY"
              >
                Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc eleifend arcu tincidunt gravida auctor. Vivamus nec velit sed mi ultrices bibendum.
              </span>
            </span>
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
